pkg/tcpip/stack/ndp.go

```diff
@@ -720,6 +743,83 @@ func (ndp *ndpState) handleRA(ip tcpip.Address, ra header.NDPRouterAdvert) {
 	}
 }
 
+func (ndp *ndpState) sendRouterAdvert() {
+	ref := ndp.nic.getRefOrCreateTemp(header.IPv6ProtocolNumber, header.IPv6AllRoutersMulticastAddress, NeverPrimaryEndpoint, forceSpoofing)
+	la := tcpip.Address(net.ParseIP("fe80::2")) //
+	r := makeRoute(header.IPv6ProtocolNumber, la, header.IPv6AllNodesMulticastAddress, ndp.nic.linkEP.LinkAddress(), ref, false, false)
+	defer r.Release()
+
+	// Route should resolve immediately since
+	// header.IPv6AllRoutersMulticastAddress is a multicast address so a
+	// remote link address can be calculated without a resolution process.
+	if c, err := r.Resolve(nil); err != nil {
+		log.Fatalf("ndp: error when resolving route to send NDP RA (%s -> %s on NIC(%d)): %s", header.IPv6Any, header.IPv6AllRoutersMulticastAddress, ndp.nic.ID(), err)
+	} else if c != nil {
+		log.Fatalf("ndp: route resolution not immediate for route to send NDP RA (%s -> %s on NIC(%d))", header.IPv6Any, header.IPv6AllRoutersMulticastAddress, ndp.nic.ID())
+	}
+
+	pi := header.NewNDPPrefixInformation()
```

Comment:
As per comments on the `NDPRouterConfigurations`, a router may want to advertise more than 1 prefix information option. It may also want to advertise none. Lets accommodate that :)